src/sqlfluff/core/linter/linter.py

```diff
@@ -483,7 +483,8 @@ def lint_fix_parsed(
         else:
             ignore_buff = []
 
-        for loop in range(loop_limit):
+        save_tree = tree
+        for loop in range(loop_limit + 1):
```

Comment:
When I changed to a `for/else` loop (see the comment below for more info), I had to adjust the loop range in order to keep the same behavior -- one of the existing tests surfaced this subtle issue.